### PR TITLE
Add watch on pod creation

### PR DESCRIPTION
**Pull request by infiniteregrets**
It looks like Job creation and pod listing happens too fast, such that the pod has not been created yet and this leads to the following panic:
```
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', mirrord-layer/src/pod_api.rs:127:34
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
`let pod = pods.items.first().unwrap();`
We use the watch API and wait till the resource has been added. https://docs.rs/kube/latest/kube/struct.Api.html#method.watch

### CODE REVIEW

File: mirrord-layer/src/pod_api.rs [hunk 1]
```diff
@@ -120,6 +122,16 @@ pub async fn create_agent(
         .unwrap();
 
     let pods_api: Api<Pod> = Api::namespaced(client.clone(), agent_namespace);
+    let params = ListParams::default().labels(&format!("job-name={}", agent_job_name));
+    let mut stream = pods_api.watch(&params, "0").await.unwrap().boxed();
+    while let Some(status) = stream.try_next().await.unwrap() {
```
Comment (aviramha) [inline mirrord-layer/src/pod_api.rs:127]: need to add timeout on that

**Review by aviramha — CHANGES_REQUESTED** [review]

File: mirrord-layer/src/pod_api.rs (hunk 1 above)
Comment (infiniteregrets) [reply mirrord-layer/src/pod_api.rs:127]: not sure but what if the resource has not been added yet and if we add a timeout, wouldnt it break out of the loop without the pod being there? the docs say the default timeout is 295 seconds though

File: mirrord-layer/src/pod_api.rs [hunk 2]
```diff
@@ -120,6 +122,16 @@ pub async fn create_agent(
...
+        match status {
+            WatchEvent::Added(_) => break,
+            WatchEvent::Error(_) => error!("Error watching pods"),
```
Comment (aviramha) [inline mirrord-layer/src/pod_api.rs:130]: break here/unwrap

File: mirrord-layer/src/pod_api.rs (hunk 1 above)
Comment (aviramha) [reply mirrord-layer/src/pod_api.rs:127]: well, yes but we don't want it to wait indefinitely

**Review by aviramha — CHANGES_REQUESTED** [review]

File: mirrord-layer/src/pod_api.rs (hunk 1 above)
Comment (infiniteregrets) [reply mirrord-layer/src/pod_api.rs:127]: added a 10 second timeout

File: mirrord-layer/src/pod_api.rs (hunk 2 above)
Comment (infiniteregrets) [reply mirrord-layer/src/pod_api.rs:130]: not sure wym, where?

File: mirrord-layer/src/pod_api.rs [hunk 3]
```diff
@@ -120,6 +122,21 @@ pub async fn create_agent(
...
 
     let pods_api: Api<Pod> = Api::namespaced(client.clone(), agent_namespace);
+    let params = ListParams::default()
+        .labels(&format!("job-name={}", agent_job_name))
+        .timeout(10);
+    let mut stream = pods_api.watch(&params, "0").await.unwrap().boxed();
+    while let Some(status) = stream.try_next().await.unwrap() {
+        match status {
+            WatchEvent::Added(_) => break,
+            WatchEvent::Error(_) => {
```
Comment (aviramha) [inline mirrord-layer/src/pod_api.rs:132]: capture the error variable and add it to the error macro

**Review by aviramha — COMMENTED** [review]
almost there!

**Review by aviramha — APPROVED** [review]
Awesome, Thanks!

**Pull request merged**